### DISCUSSION TITLE
I have a ceiling light, I reversed the payload but I would need guidance on writing the device description yaml

### DISCUSSION THREAD

**Discussion by NicolasMarcotte** [make-all/tuya-local#1405] [Q&A]
My ceiling light is composed of 1 white and 1 colors light that can be controlled separatetly by tuya apps, so in home assistant I would like to represent it as 2 devices.

here is the "Mix light" payload composition in bytes:

```
00 0A  HH HH SS SS LL LL WB WB WT WT
where
A: b00ZY (Z : color on/off  and Y : white on/off)
HH HH: Hue 0 to 360 in hex
SS SS: Saturation 0 to 1000 in hex
LL LL: Luma 0 to 1000 in hex
WB WB : white brigtness 0 to 1000 in hex
WT WT : white temperature 0 to 1000 in hex

```

The tuya app first send a "Mix Light" DP id:51, and if the light is off send a on on the DP id:20. I know how to implements this directly in python, but I have no Idea on how, or even if it possible, to accomplish this in yaml. Could someone help me ?

**Comment by illuzn** [comment]
Based on the (current documentation)[https://github.com/make-all/tuya-local/blob/main/custom_components/tuya_local/devices/README.md] and the information you've provided, this isn't possible via YAML. 

- Tuya-local doesn't support setting up 2 devices (as opposed to entities) from 1 tuya device.

- Every device in tuya-local requires a primary entity and that primary entity must have a dps set.

The situation might be different if there was another dps (say dps 1 is a simple on/off switch).

In order to represent this:
- I would make the primary entity a switch which maps to dps 1.
- I would make 2 secondary light entities:
      - the on/off would mirror or redirect the primary entity switch in dps 1. Each individual bulb being on/off would be set by the Luma/ Brightness attribute you describe above.
      - one of them would have a bitfield or hex that represents what you've got up there.
      - you would probably have to code a new mapping to map part of a bitfield or hex to the second light entity.

If you know how to do this in python. It's probably easier to write your own custom integration using tinytuya (the uptodate fork of pytuya) to do the heavy lifting. I'd use the CoordinatorEntity built in to HomeAssistant to coordinate the data push between the two light entities.

**Comment by Peerke1** [comment]
maybe the idea below descibes your device. Mine has a white light and a Halo (that points to the ceiling). both are controlled independantly in the app while the dp51 frame targets both. 

i analysed the app - light communication and made the idea below:
https://github.com/make-all/tuya-local/discussions/5825
